### PR TITLE
Fix exception mapping from JS to dart

### PR DESCRIPTION
**Pull request by annagrin**
A prior SDK change introduced a regression - exception mapping stopped working due detection in DWDS relying on type signatures for exceptions.

Allow all subtypes of `NativeError` as exceptions.

Closes: https://github.com/dart-lang/webdev/issues/1821

### CODE REVIEW

File: frontend_server_common/pubspec.yaml [hunk 1]
```diff
@@ -6,7 +6,10 @@ environment:
   sdk: ">=3.0.0-134.0.dev <4.0.0"
 
 dependencies:
-  dwds: 18.0.0
+  # Uncomment before releasing webdev
```
Comment (elliette) [inline frontend_server_common/pubspec.yaml:9]: Remove? (here and below)

**Review by elliette — APPROVED** [review]
Approved with one comment to resolve

**Review by nshahan — APPROVED** [review]

File: frontend_server_common/pubspec.yaml (hunk 1 above)
Comment (annagrin) [reply frontend_server_common/pubspec.yaml]: done!

File: dwds/lib/src/debugging/metadata/class.dart [hunk 2]
```diff
@@ -20,6 +22,31 @@ final classRefForString = classRefFor(_dartCoreLibrary, InstanceKind.kString);
...
+///
+/// TODO(annagrin): this breaks on name changes for JS types.
+/// https://github.com/dart-lang/sdk/issues/51583
+final classRefForNativeJsError =
+    classRefFor(_dartInterceptorsLibrary, 'NativeError');
+
+/// Returns true for non-dart JavaScript classes.
+///
+/// TODO(annagrin): this breaks on name changes for JS types.
+/// https://github.com/dart-lang/sdk/issues/51583
```
[the remark this answers is not in the document]
Comment (annagrin) [reply dwds/lib/src/debugging/metadata/class.dart:39]: Totally agree.

**Pull request merged**